src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java

```diff
@@ -399,8 +404,110 @@ void getHearingLocations() {
     @DisplayName("When override Hearing Venue Epims Ids is not empty getHearingLocations returns the override values")
     @Test
     void getHearingLocationsOverride() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
 
-        SscsCaseData caseData = SscsCaseData.builder()
+        List<HearingLocation> result = HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder);
+
+        assertThat(result)
+            .hasSize(2)
+            .extracting("locationId","locationType")
+            .containsExactlyInAnyOrder(
+                tuple("219164", COURT),
+                tuple("436578", COURT));
+    }
+
+    @DisplayName("When a case has been adjourned and a different venue has been selected, return the new venue")
+    @Test
+    void getHearingLocationsAdjournmentNewVenue() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
+
+        caseData.getAppeal().setHearingSubtype(HearingSubtype.builder()
+                .wantsHearingTypeTelephone("yes")
+                .hearingTelephoneNumber("07483871426").build());
+        caseData.getAppeal().setHearingOptions(HearingOptions.builder().wantsToAttend("yes").build());
+
+        DynamicListItem item = new DynamicListItem("219164", "");
+        DynamicList list = new DynamicList(item, Collections.emptyList());
+
+        caseData.setAdjournCaseNextHearingVenue("somewhereElse");
+        caseData.setAdjournCaseNextHearingVenueSelected(list);
+
+        List<HearingLocation> result = HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder);
+
+        assertThat(result)
+            .hasSize(1)
+            .extracting("locationId","locationType")
+            .containsExactlyInAnyOrder(tuple("219164", COURT));
```

Comment:
This whole test can be extracted to a helper method and used again in the test below - getHearingLocationsAdjournmentSameVenue

That being said, are the two tests different enough? I can't see where the somewhereElse venue is selected.

```suggestion

void checkResult(String nextHearingVenue) {
        SscsCaseData caseData = buildOverrideHearingLocations();

        caseData.getAppeal().setHearingSubtype(HearingSubtype.builder()
                .wantsHearingTypeTelephone("yes")
                .hearingTelephoneNumber("07483871426").build());
        caseData.getAppeal().setHearingOptions(HearingOptions.builder().wantsToAttend("yes").build());

        DynamicListItem item = new DynamicListItem("219164", "");
        DynamicList list = new DynamicList(item, Collections.emptyList());

        caseData.setAdjournCaseNextHearingVenue("nextHearingVenue");
        caseData.setAdjournCaseNextHearingVenueSelected(list);

        List<HearingLocation> result = HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder);

        assertThat(result)
            .hasSize(1)
            .extracting("locationId","locationType")
            .containsExactlyInAnyOrder(tuple("219164", COURT));
}
            
```

Reply:
I think there are enough tests, we check what happens when somewhere else is selected or where same value is selected as well as checking both throw conditions. Also previous tests pick up on when the hearing hasn't been adjourned

Reply:
Also they have different setups, I've split up parts of the setup into their own helper methods so hopefully this will make it more clear

Reply:
ok